x/leveragelp/keeper/msg_server_add_collateral_test.go

```diff
@@ -0,0 +1,227 @@
+				Creator:    addresses[0].String(),
+				Id:         1,
+				Collateral: sdk.NewInt(0),
+			},
+			true,
+			types.ErrPositionDoesNotExist.Error(),
+			func() {
+				suite.ResetSuite()
+				SetupCoinPrices(suite.ctx, suite.app.OracleKeeper)
+				initializeForAddCollateral(suite, addresses, asset1, asset2, false)
+			},
+		},
+		{"pool not found",
+			&types.MsgAddCollateral{
+				Creator:    addresses[0].String(),
+				Id:         1,
+				Collateral: sdk.NewInt(0),
+			},
+			true,
+			types.ErrPoolDoesNotExist.Error(),
+			func() {
+				suite.ResetSuite()
+				SetupCoinPrices(suite.ctx, suite.app.OracleKeeper)
+				initializeForAddCollateral(suite, addresses, asset1, asset2, true)
+				openPosition(suite, addresses[0], collateralAmount, leverage)
+				suite.app.LeveragelpKeeper.DeletePool(suite.ctx, 1)
+			},
+		},
+		{"pool not enabled",
+			&types.MsgAddCollateral{
+				Creator:    addresses[0].String(),
+				Id:         1,
+				Collateral: sdk.NewInt(0),
+			},
+			true,
+			types.ErrPositionDisabled.Error(),
+			func() {
+				suite.ResetSuite()
+				SetupCoinPrices(suite.ctx, suite.app.OracleKeeper)
+				initializeForAddCollateral(suite, addresses, asset1, asset2, true)
+				openPosition(suite, addresses[0], collateralAmount, leverage)
+				pool, _ := suite.app.LeveragelpKeeper.GetPool(suite.ctx, 1)
+				pool.Enabled = false
+				suite.app.LeveragelpKeeper.SetPool(suite.ctx, pool)
+			},
+		},
+		{"repaying more than allowed",
+			&types.MsgAddCollateral{
+				Creator:    addresses[0].String(),
+				Id:         1,
+				Collateral: sdk.NewInt(math.MaxInt64),
+			},
+			true,
+			types.ErrInvalidCollateral.Error(),
+			func() {
+				suite.ResetSuite()
+				SetupCoinPrices(suite.ctx, suite.app.OracleKeeper)
+				initializeForAddCollateral(suite, addresses, asset1, asset2, true)
+				openPosition(suite, addresses[0], collateralAmount, leverage)
+			},
+		},
+		{"balance too low to repay",
+			&types.MsgAddCollateral{
+				Creator:    addresses[0].String(),
+				Id:         1,
+				Collateral: sdk.NewInt(10000),
+			},
+			true,
+			"spendable balance  is smaller",
+			func() {
+				suite.ResetSuite()
+				SetupCoinPrices(suite.ctx, suite.app.OracleKeeper)
+				initializeForAddCollateral(suite, addresses, asset1, asset2, true)
+				openPosition(suite, addresses[0], collateralAmount, leverage)
+				amount := suite.app.BankKeeper.GetBalance(suite.ctx, addresses[0], "uusdc")
+				err := suite.app.BankKeeper.SendCoins(suite.ctx, addresses[0], addresses[1], sdk.NewCoins(amount))
+				if err != nil {
+					panic(err)
+				}
+
+			},
+		},
+		{"positive case",
+			&types.MsgAddCollateral{
+				Creator:    addresses[0].String(),
+				Id:         1,
+				Collateral: sdk.NewInt(10000),
+			},
+			false,
+			"",
+			func() {
+				suite.ResetSuite()
+				SetupCoinPrices(suite.ctx, suite.app.OracleKeeper)
+				initializeForAddCollateral(suite, addresses, asset1, asset2, true)
+				openPosition(suite, addresses[0], collateralAmount, leverage)
+			},
```

Comment:
is it possible to verify here if collateral amount has increased and lpTotal amount remains same after adding collateral ?

Reply:
Yes, let me change the unit test structure and add post validations